opencsp/app/sofast/lib/BlobIndex.py

```diff
@@ -19,6 +19,29 @@ class Step(Enum):
     LEFT_OR_UP = -1
 
 
+class DebugBlobIndex:
+    """Debug class for BlobIndex
+
+    Parameters
+    ----------
+        debug_active : bool
+            Flag to turn on debugging, default false
+        figures : list[Figure]
+            Container to hold debug figures
+        bg_image : ndarray
+            2d ndarray, by default None. Background image (i.e. the image with the blobs present)
+            to plot in debugging plots. If None, background image is not plotted.
+        name : str
+            Name of current instance to prepend to plot titles
+    """
+
+    def __init__(self):
+        self.debug_active: bool = False
+        self.figures: list = []
+        self.bg_image: np.ndarray = None
+        self.name: str = ''
+
```

Comment:
I suggest adding a destructor method that closes all the figures held onto by this instance. Maybe consider extending https://github.com/sandialabs/OpenCSP/blob/develop/opencsp/common/lib/render/lib/AbstractPlotHandler.py for this functionality.

Reply:
Done.